reporter/aspec/src/test/java/org/itsallcode/openfasttrace/report/aspec/TestASpecReport.java

```diff
@@ -0,0 +1,732 @@
+                if( values.size() > 4 )
+                {
+                    b.append( ".*<transitiveCoverageStatus>" ).append( values.get( 4 ) ).append( "</transitiveCoverageStatus>" );
+                }
+                if( values.size() > 5 )
+                {
+                    b.append( ".*<coveringStatus>" ).append( values.get( 5 ) ).append( "</coveringStatus>" );
+                }
+                b.append( ".*</" ).append( fieldName ).append( ">" );
+            }
+            b.append( ".*</" ).append( fieldWrapper ).append( ">" );
+        }
+    }
+
+    private static class Field
+    {
+        enum Type
+        {
+            ID, VERSION, STATUS, DESCRIPTION, NEEDS_COVERAGE, SHALLOW_COVERAGE, TRANSITIVE_COVERAGE, COVERED_ITEMS,
+            COVERED_TYPES, UNCOVERED_TYPES, COVERING_ITEMS, DEPENDENCIES
+        }
+
+        private final Type type;
+        private final List<String> values = new ArrayList<>();
+
+        public Field( final Type type, final String... value )
+        {
+            this.type = type;
+            Collections.addAll( values, value );
+        }
+
+        public Field( final Type type, final int value )
+        {
+            this.type = type;
+            values.add( Integer.toString( value ) );
+        }
+
+        public Field( final Type type, final ItemStatus value )
+        {
+            this.type = type;
+            values.add( value.toString() );
+        }
+
+        public Field( final Type type, final SampleArtifactTypes... value )
+        {
+            this.type = type;
+            Arrays.stream( value ).forEach( entry -> values.add( entry.toString() ) );
+        }
+
+        public Field( final Type type, final DeepCoverageStatus value )
+        {
+            this.type = type;
+            values.add( value == DeepCoverageStatus.COVERED ? "COVERED" : "UNCOVERED" );
+        }
+
+        public Field( final Type type, final String id, final int version, ItemStatus status,
+                      final DeepCoverageStatus ownCoverage,
+                      final DeepCoverageStatus transitiveCoverage,
+                      final ASpecReport.CoveringStatus coveringStatus )
+        {
+            this.type = type;
+            values.add( id );
+            values.add( Integer.toString( version ) );
+            values.add( status.toString() );
+            values.add( ownCoverage == DeepCoverageStatus.COVERED ? "COVERED" : "UNCOVERED" );
+            values.add( transitiveCoverage == DeepCoverageStatus.COVERED ? "COVERED" : "UNCOVERED" );
+            values.add( coveringStatus.getLabel() );
+        }
+
+        public Field( final Type type, final String id, final int version )
+        {
+            this.type = type;
+            values.add( id );
+            values.add( Integer.toString( version ) );
+        }
+
+        public List<String> getValues()
+        {
+            return values;
+        }
+    }
+
+    static class StringRegexp extends SubstringMatcher
```

Comment:
This could go into the `openfasttrace-testutil` module?

Reply:
Done